docs/tasks/debug-application-cluster/logging-stackdriver.md

```diff
@@ -5,16 +5,94 @@ assignees:
+**Note:** By default, only container's standard output and standard error will be
+collected by logging solution. To collect logs written by an application to
+a file, you can take a look at the [sidecar approach](/docs/concepts/cluster-administration/logging/#using-a-sidecar-container-with-the-logging-agent).
 
-The following guide describes gathering a container's standard output and standard error. To gather logs written by an application to a file, you can use [a sidecar approach](https://github.com/kubernetes/contrib/blob/master/logging/fluentd-sidecar-gcp/README.md).
+## Deploying
 
-## Overview
+In order to ingest logs, as explained in the logging overview, logging agent
+should be deployed to each node. Stackdriver logging agent is configured
+fluentd instance. Configuration for fluentd is stored in ConfigMap and
+instances are managed using DaemonSet, which makes Kubernetes schedule
+an agent on each node. The way how ConfigMap and DaemonSet are deployed
+in a cluster depends on your setup.
 
-After creation, you can discover logging agent pods in the `kube-system` namespace,
-one per node, by running the following command:
+### GKE
+
+Stackdriver is a default logging solution for clusters on GKE. Unless you
+deliberately opt out of using Stackdriver Logging, it will be deployed
+and working out of the box without any further actions.
+
+### Deploying with a new cluster
+
+**Note:** this section is only valid if you're using `kube-up.sh` for starting
+your cluster. Otherwise, please refer to the next section.
+
+Before starting your cluster, set `KUBE_LOGGING_DESTINATION` environment
+variable to `gcp` and, unless on GCE, include
+`alpha.kubernetes.io/fluentd-ds-ready=true` to`KUBE_NODE_LABELS` variable.
+Once started, cluster should have Stackdriver logging agent running on each
+node. If deployed that way, DaemonSet for the agents and ConfigMap with
+the agents configuration will be addons.
+
+### Deploying to an existing cluster
+
+1. Apply label on each node, if not present already.
+
+    Stackdriver logging agent deployment is using node labels to understand to
+    which nodes it should be allocated. If describing your nodes produces the
+    following output
+
+    ```shell
+    $ kubectl describe node $NODE_NAME
+    ...
+    Labels:         alpha.kubernetes.io/fluentd-ds-ready=true
+    ...
+    ```
+
+    then you don't need to do anything else in this step. If you can't find
+    `alpha.kubernetes.io/fluentd-ds-ready=true` among labels, execute the following
+    command for each node:
+
+    ```shell
+    $ kubectl label node $NODE_NAME alpha.kubernetes.io/fluentd-ds-ready=true
+    ```
+
+    Note that if node is recreated, label has to be applied again. Kubelet
```

Comment:
"__Note:__ If a node fails and has to be recreated, you must re-apply the label to the recreated node. To make this easier, you can use Kubelet's command-line parameter for applying node labels in your node startup script."